armsrc/iclass.c

```diff
@@ -2152,3 +2152,216 @@ void iClass_Restore(iclass_restore_req_t *msg) {
         reply_ng(CMD_HF_ICLASS_RESTORE, isOK, NULL, 0);
     }
 }
+
+void generate_single_key_block_inverted(const uint8_t startingKey[PICOPASS_BLOCK_SIZE], uint32_t index, uint8_t keyBlock[PICOPASS_BLOCK_SIZE]) {
+    uint32_t carry = index;
+    memcpy(keyBlock, startingKey, PICOPASS_BLOCK_SIZE);
+
+    for (int j = PICOPASS_BLOCK_SIZE - 1; j >= 0; j--) {
+        uint8_t increment_value = carry & 0x07;  // Use only the last 3 bits of carry
+        keyBlock[j] = increment_value;  // Set the last 3 bits, assuming first 5 bits are always 0
+
+        carry >>= 3;  // Shift right by 3 bits for the next byte
+        if (carry == 0) {
+            // If no more carry, break early to avoid unnecessary loops
+            break;
+        }
+    }
+}
+// Function to convert an unsigned int to binary string
+void intToBinary(unsigned int num, char *binaryStr, int size) {
```

Comment:
we use  `uintX_t`  to denote unsigned it.  
And functions like these should be reused and located in utils.c

Reply:
Fixed unsigned int, will check for utils.c